invenio_vocabularies/services/service.py

```diff
@@ -135,3 +137,35 @@ def search(self, identity, params=None, es_preference=None, type=None,
             }),
             links_item_tpl=self.links_item_tpl,
         )
+
+    def read_all(self, identity, fields, type, cache=True, **kwargs):
+        """Search for records matching the querystring."""
+        cache_key = "-".join(fields)
+        results = current_cache.get(cache_key)
+        es_query = Q("match_all")
+
+        # If not found, NoResultFound is raised (caught by the resource).
+        vocabulary_type = VocabularyType.query.filter_by(id=type).one()
```

Comment:
This will perform a database query and should be avoided when the results are cached.

Reply:
we wont be able to reconstruct the search fully then... but since there are no links it should be potentially fine...